cmd/gossamer/init.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/trie"
+)
+
+func loadTrie(t *trie.Trie, data []map[string]string) error {
+	for _, field := range data {
+		for key, value := range field {
+			keyBytes, err := common.HexToBytes(key)
+			if err != nil {
+				return err
+			}
+			valueBytes, err := common.HexToBytes(value)
+			if err != nil {
+				return err
+			}
+			err = t.Put(keyBytes, valueBytes)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func commitToDb(t *trie.Trie) error {
```

Comment:
I'm a bit confused why this is also commits to the trie. I feel like trie commit should happen above and then you should be calling `t.WriteToDb()` directly (possibly also in `loadTrie` if that's how you choose to define it)

Reply:
what do you mean? this just writes the trie data to the DB after it's been loaded, and it's called in  `loadGenesis` not `loadTrie`
also the functions might be named weirdly, `t.WriteToDb` writes the trie to the batch writer then `t.Commit` actually finishes writing the batch to the DB

Reply:
it's just that its `commitToDb` not `commitToDbAndTrie`. I feel like `loadTrie` should commit the state to the trie, a trie with no committed state isn't fully loaded.

If that reduces this function to just writing to db then it should be removed altogether, and we should call `t.writeToDB()` directly.

Reply:
okay, I tried moving the contents of commitToDB to loadTrie and it causes a nil pointer error for some reason. will try to fix

Reply:
ok I have moved the commit to DB into `loadGenesis` and removed this function